### PR TITLE
Upgrade to be terraform .12 compatible & other changes

### LINKED ISSUE #5: alarm names should be unique
terraform doesn't complain about the conflict, but the alarms will constantly have their dimension overwritten with every deploy if the names aren't unique.

### LINKED ISSUE #16: update to be hcl2 compliant
Using terraform 0.12.20, seeing this error which implies this file is not HCL2 compliant.
```
Error: Unsupported block type

  on .terraform/modules/rds_alarms/alarms.tf line 80, in resource "aws_cloudwatch_metric_alarm" "disk_queue_depth_too_high":
  80:   dimensions {

Blocks of type "dimensions" are not expected here. Did you mean to define
argument "dimensions"? If so, use the equals sign to assign it a value.
````

### PR DESCRIPTION
**Pull request by caleb15**
* Upgrades code to be terraform .12 compatible fixes #16
* Makes alarm names unique fixes #5 
* __Breaking change__: requires sns topic param instead of creating one for the user. The user may already have a prexisting SNS topic that they might want to use. The repo should just concern itself with alarms and not with SNS.
* __Breaking change__: `db_instance_id` changed to `db_instance_ids` to support multiple db's
* add `alarm_period` and `alarm_evaluation_periods` params

We are running this change in production for [15Five](http://15five.com/).

### CODE REVIEW

**Comment by jamengual** [comment]
> It shouldn't be assumed that it's okay to create a new topic - the user might want to reuse an existing sns topic for the RDS alarms. DEV-812

That is going to break other modules, could it be possible for you to use 
```
count = length(var.aws_sns_topic_arn) != 0 ? 1 : 0
```
to the topics so that if there is no `var.aws_sns_topic_arn` supplied it created one for you?

thanks

**Comment by caleb15** [comment]
@jamengual Yeah, that should be possible. But either way the PR will still have a breaking change with the db instance id param.

I'm going to wait for a maintainer to respond to see what changes would be necessary to merge this in.

**Comment by jamengual** [comment]
for the breaking changes I think is possible to maybe leave them empty or use a dynamic.

**Comment by nitrocode** [comment]
To make this easier to review, could you put these changes in separate PRs?

We should at the very least have a separate PR for terraform11 to terraform12 changes to make this easier to review.

File: main.tf
Comment (3h4x) [inline main.tf:10]: What is the reason behind removing other categories? 
I think it would be best to preserve previous list as a default. To customize you can create `event_categories` variable of `list(string)` type.

File: variables.tf
Comment (3h4x) [inline variables.tf:69]: empty line

File: main.tf
Comment (3h4x) [inline main.tf:4]: Definitely we need a name prefix so all resources with names created by this module are aligned. Additionally I think `rds-sns-alarms` is more telling.
```suggestion
  name_prefix = "${var.name_prefix}rds-sns-alarms"
```

File: variables.tf
Comment (3h4x) [inline variables.tf:13]: "The bla"? I guess you mean ARN

File: alarms.tf
Comment (3h4x) [inline alarms.tf:20]: unit = "Percent"

File: alarms.tf
Comment (3h4x) [inline alarms.tf:39]: unit = "Percent"

File: alarms.tf
Comment (3h4x) [inline alarms.tf:58]: unit = "Count"

File: alarms.tf
Comment (3h4x) [inline alarms.tf:77]: unit = "Count"

File: alarms.tf
Comment (3h4x) [inline alarms.tf:96]: unit = "Bytes"

File: alarms.tf
Comment (3h4x) [inline alarms.tf:116]: unit = "Bytes"

File: alarms.tf
Comment (3h4x) [inline alarms.tf:135]: unit = "Bytes"

File: alarms.tf
Comment (3h4x) [inline alarms.tf]: Please add `db_instance_id` to every alarm_description.

**Review by 3h4x — CHANGES_REQUESTED** [review]
Great work @caleb15 ! Thank you very much for PR.
These are great changes. I'm in favour of removing SNS topic creation without any backward compatibility. It should be fairly easy to migrate to new version thanks to variable `aws_sns_topic_arn`. In @cloudposse/contributors we are working on SNS module (initial PR already pushed) that should be used in conjunction with any CloudWatch alarm module.
Additionally it's a good change to monitor multiple RDS instances with this single module. We can avoid backward incompatibilty by reducing readablity of code and by having tech debt. Let's avoid that.
I left couple of questions and few improvements.  

PS: I'm going to push `pre-commit` to this repo for `tf fmt` and other lints. Please merge it to your PR when it's done.

**Comment by caleb15** [comment]
Sorry for the delay on responding to the PR review, I've created a ticket so I remember to get to it eventually. It might be a while. Thanks for the review!

**Comment by awartoft** [comment]
@caleb15 Any chance we might see an update :)?

**Comment by caleb15** [comment]
@macnibblet yes, but it may take a while :see_no_evil: I have the email starred in my inbox along with 30 others.

File: main.tf
Comment (jamengual) [inline main.tf:13]: why all this have to be deleted?

**Comment by tovbinm** [comment]
@caleb15 how far are you from finishing this one up?

**Comment by adamantike** [comment]
Also waiting for this PR to be merged, to stop using a fork of this repo!

**Comment by Gowiem** [comment]
@maximmi are you still the owning migration to 0.12 PRs? Any chance you could check this one out?

**Review by maximmi — CHANGES_REQUESTED** [review]
@caleb15 thank you for contribution
we have our standards for Terraform 0.12+ code, so that tests and github workflows should be in place. It might be hard to implement such things by you, so I will try to update your PR to respect all requirements.
For the reference you could pick any our repo with terraform 0.12+ support, for example: https://github.com/cloudposse/terraform-aws-s3-bucket

**Pull request closed without merging**
